### PR TITLE
Tab

### PR DESCRIPTION
**Pull request by nicosampler**
This PR adds the Tab component

### CODE REVIEW

File: src/navigation/Tab/index.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+const TabWrapper = styled.div<{ variantStyle: string }>`
+  border-bottom: ${({ variantStyle }) =>
+    variantStyle === 'outlined'
+      ? '1px solid ' + theme.colors.overlay.color
+      : 'none'};
+`;
+
+interface CustomTabsProps extends TabsProps {
+  variantStyle: string;
+  children: any;
```
Comment (mikheevm) [inline src/navigation/Tab/index.tsx:37]: please don't use any

File: src/navigation/Tab/index.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+}
+const CustomTabs = ({ variantStyle, ...rest }: CustomTabsProps): any => {
```
Comment (mikheevm) [inline src/navigation/Tab/index.tsx:39]: same here

File: src/navigation/Tab/index.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  })(TabsMui);
+
+  return <CustomTabsMui {...rest} />;
+};
+
+interface CustomTabProps extends TabProps {
+  variantStyle: string;
+}
+
+const CustomTab = ({ variantStyle, ...rest }: CustomTabProps): any => {
```
Comment (mikheevm) [inline src/navigation/Tab/index.tsx:76]: same here

File: src/navigation/Tab/index.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      },
+      textTransform: variantStyle === 'contained' ? 'capitalize' : 'uppercase'
+    }
+  })(TabMui);
+
+  return <CustomTabMui {...rest} />;
+};
+
+function Tab({ onChange, items, selectedTab, variant = 'outlined', fullWidth }: Props) {
+  const handleChange = (_event: React.ChangeEvent<{}>, value: any) => {
```
Comment (mikheevm) [inline src/navigation/Tab/index.tsx:97]: same here

File: src/navigation/Tab/index.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+      </Text>
+    );
+  };
+
+  return (
+    <TabWrapper variantStyle={variant}>
+      <CustomTabs
+        variant={fullWidth ? 'fullWidth' : 'scrollable'}
+        value={selectedTab}
+        onChange={handleChange as any}
```
Comment (mikheevm) [inline src/navigation/Tab/index.tsx:130]: same here

File: src/navigation/Tab/index.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+export type Props = {
+  onChange: (selectedIndex: string) => void;
+  items: Array<Item>;
+  selectedTab: string;
+  variant?: 'outlined' | 'contained';
+  fullWidth?: Boolean;
+};
+
+const TabWrapper = styled.div<{ variantStyle: string }>`
```
Comment (mikheevm) [inline src/navigation/Tab/index.tsx:28]: why a different name (it gets passed as `variant`)? also the type is incorrect

```
variant?: 'outlined' | 'contained';
```

File: src/navigation/Tab/index.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  border-bottom: ${({ variantStyle }) =>
+    variantStyle === 'outlined'
+      ? '1px solid ' + theme.colors.overlay.color
```
Comment (mikheevm) [inline src/navigation/Tab/index.tsx:31]: I think theme gets passed as props.theme, doesn't it?

**Comment by mikheevm** [comment]
Please change project's typescript compiler rules so it doesn't allow usage of `any`

**Comment by nicosampler** [comment]
> Please change project's typescript compiler rules so it doesn't allow usage of `any`

I'll do it in another PR and fix all the any

File: src/navigation/Tab/index.tsx (hunk 5 above)
Comment (nicosampler) [reply src/navigation/Tab/index.tsx:138]: This is the only `any` I couldn't figure it out how to resolve, onChange does not recognize the type assigned to the function. If you can, take a look, please.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr13--safereactcomponents.review.gnosisdev.com

File: src/navigation/Tab/index.tsx (hunk 5 above)
Comment (mikheevm) [reply src/navigation/Tab/index.tsx]: @nicosampler done. this was a little bit tricky because it uses generics & type arguments, hope I got it right

Comment (mikheevm) [reply src/navigation/Tab/index.tsx]: also bumped dependencies for tsc/material-ui

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr13--safereactcomponents.review.gnosisdev.com

File: src/navigation/Tab/index.tsx [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        backgroundColor: variantStyle === 'contained' ? 'none' : 'inherit'
+      },
+      textTransform: variantStyle === 'contained' ? 'capitalize' : 'uppercase'
+    }
+  })(TabMui);
+
+  return <CustomTabMui {...rest} />;
+};
+
+function Tab({
```
Comment (mikheevm) [inline src/navigation/Tab/index.tsx:105]: why are we mixing const & function?

Comment (nicosampler) [reply src/navigation/Tab/index.tsx]: changed

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr13--safereactcomponents.review.gnosisdev.com

File: src/navigation/Tab/index.tsx (hunk 7 above)
Comment (mikheevm) [reply src/navigation/Tab/index.tsx:33]: Probably I was wrong about that. So is our goal to make the components standalone? E.g props.theme is not available (ThemeProvider is not used). If yes, your usage was correct

**Pull request merged**